batch/src/main/java/org/openmrs/analytics/JdbcFhirMode.java

```diff
@@ -0,0 +1,233 @@
+		dataSource.setPassword(options.getDbPassword());
+		dataSource.setMaxPoolSize(options.getJdbcMaxPoolSize());
+		dataSource.setInitialPoolSize(10); // TODO: make this configurable if issues arises
+		return JdbcIO.DataSourceConfiguration.create(dataSource);
+	}
+	
+	public static PCollection<KV<String, Iterable<Integer>>> createChunkRanges(String tableName, int fetchSize, Pipeline p,
+	        JdbcIO.DataSourceConfiguration config) {
+		String tableId = tableName + "_id";
+		return p.apply(String.format("Read Max table Index Column for: %s", tableName),
+		    JdbcIO.<String> read().withDataSourceConfiguration(config)
+		            .withQuery(String.format("SELECT MAX(`%s`) from %s", tableId, tableName))
+		            .withRowMapper(new JdbcIO.RowMapper<String>() {
+			            
+			            @Override
+			            public String mapRow(ResultSet resultSet) throws Exception {
+				            return resultSet.getString(1);
+			            }
+		            }).withOutputParallelization(false).withCoder(StringUtf8Coder.of()))
+		        .apply("Distribute", ParDo.of(new DoFn<String, KV<String, Integer>>() {
+			            
+			        @ProcessElement
+			        public void processElement(ProcessContext c) {
+				        int count = Integer.parseInt(c.element());
+				        int ranges = count / fetchSize;
+				        for (int i = 0; i < ranges; i++) {
+					        int rangeFrom = i * fetchSize;
+					        int rangeTo = (i + 1) * fetchSize;
+					        String range = String.format("%s,%s", rangeFrom, rangeTo);
+					        c.output(KV.of(range, rangeFrom));
+				        }
+				        if (count > ranges * fetchSize) {
+					        int rangeFrom = ranges * fetchSize;
+					        int rangeTo = ranges * fetchSize + count % fetchSize;
+					        String range = String.format("%s,%s", rangeFrom, rangeTo);
+					        c.output(KV.of(range, rangeFrom));
+				        }
+			        }
+		        })).apply("Group By", GroupByKey.create()); //necessary for returning KV<String, Iterable<Integer>>
+	}
+	
+	// Uuid might be missing in child table, to solve this issue we are using UuidTable field mappings in json config
+	// TODO: A better solution is to Use join clause to get parent table, See comments left here:
+	//  https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/issues/46#issuecomment-728979138
+	public static PCollection<String> generateFhirUrl(String tableName, String resourceType, int fetchSize,
+	        PCollection<KV<String, Iterable<Integer>>> ranges, JdbcIO.DataSourceConfiguration config) {
+		String tableId = tableName + "_id";
+		return ranges.apply(String.format("Read UUIDs from %s", tableName),
+		    JdbcIO.<KV<String, Iterable<Integer>>, String> readAll().withDataSourceConfiguration(config)
+		            .withFetchSize(fetchSize).withCoder(StringUtf8Coder.of())
+		            .withParameterSetter(new JdbcIO.PreparedStatementSetter<KV<String, Iterable<Integer>>>() {
```

Comment:
This is hard to read; can you please move the anonymous class into a separate name class and use that here?

Reply:
Noted